client/blocks/upload-image/index.jsx

```diff
@@ -157,6 +272,14 @@ class UploadImage extends Component {
 			);
 		}
 
+		if ( typeof uploadingDoneContent === 'undefined' && uploadedImage && uploadedImage.URL ) {
+			uploadingDoneContent = (
+				<div className="upload-image__uploading-done-container">
+					<img src={ uploadedImage.URL } />
+				</div>
+			);
```

Comment:
The `placeholderContent` and `uploadingContent` values are created on every `render` unless a custom value is provided in `props`. One or both of them is then thrown away without being used -- depending on how the conditions with `isUploading` are evaluated.

`uploadingDoneContent` is probably never thrown away like this (can `isUploading` be true and `uploadedImage` non-null at any moment?), but only because the same (or similar) condition is evaluated twice at two places. I think this can be made a bit better.

Yes, the JS code will still be always parsed, but we can avoid _executing_ it.